tests/unit/amo/components/TestAddonReviewList.js

```diff
@@ -43,47 +32,223 @@ describe('amo/components/AddonReviewList', () => {
   describe('<AddonReviewListBase/>', () => {
     function render({
       addon = fakeAddon,
+      dispatch = sinon.stub(),
+      errorHandler = new ErrorHandler({
+        id: 'some-error-handler',
+        dispatch: sinon.stub(),
+      }),
       params = {
         addonSlug: fakeAddon.slug,
       },
       reviews = [fakeReview],
       ...customProps
     } = {}) {
       const loadedReviews = reviews ? getLoadedReviews({ reviews }) : null;
-      const { store } = createStore();
       const props = {
         addon: addon && denormalizeAddon(addon),
+        dispatch,
+        errorHandler,
+        i18n: getFakeI18nInst(),
         location: { query: {} },
         params,
         reviewCount: loadedReviews && loadedReviews.length,
         reviews: loadedReviews,
         ...customProps,
       };
 
-      const AddonReviewList = translate({ withRef: true })(AddonReviewListBase);
-      const tree = renderIntoDocument(
-        <Provider store={store}>
-          <I18nProvider i18n={getFakeI18nInst()}>
-            <AddonReviewList {...props} />
-          </I18nProvider>
-        </Provider>
-      );
-
-      return tree;
-    }
-
-    function renderToDOM(...args) {
-      return findDOMNode(render(...args));
+      return shallow(<AddonReviewListBase {...props} />);
     }
 
     it('requires an addonSlug property', () => {
       expect(() => render({ params: { addonSlug: null } }))
         .toThrowError(/addonSlug cannot be falsey/);
     });
 
+    it('waits for an addon and reviews to load', () => {
+      const root = render({ addon: null, reviews: null });
+      expect(root.find('.AddonReviewList-header-icon img').prop('src'))
+        .toContain('default');
+      expect(root.find('.AddonReviewList-header-text').find(LoadingText))
+        .toHaveLength(2);
+
+      // Make sure four review placeholders were rendered.
+      expect(root.find('.AddonReviewList-li')).toHaveLength(4);
+      // Do a sanity check on the first placeholder;
+      expect(root.find('.AddonReviewList-li h3').at(0).find(LoadingText))
+        .toHaveLength(1);
+      expect(root.find('.AddonReviewList-li p').at(0).find(LoadingText))
+        .toHaveLength(1);
+      expect(root.find('.AddonReviewList-by-line').at(0).find(LoadingText))
+        .toHaveLength(1);
+    });
+
     it('waits for reviews to load', () => {
-      const root = renderToDOM({ reviews: null });
-      expect(root.textContent).toEqual('Loading...');
+      const root = render({ addon: fakeAddon, reviews: null });
+
+      // Make sure Paginate is not rendered since it will throw an
+      // error about an empty reviewCount.
+      expect(root.find(Paginate)).toHaveLength(0);
+
+      // The other placeholders are checked in the test above.
+    });
+
+    it('fetches an addon if needed', () => {
```

Comment:
I feel like Sagas and dispatching actions that make our requests make it easier to do things like "request an add-on for this review page because neither are loaded". This all feels easier to grok than the asyncConnect stuff, and more unified in its design. 👍